Fixed #36060 -- Prevented IntegrityError in bulk_create() with order_with_respect_to. (#19012)
django/django: docs/ref/models/querysets.txt

  retrieved on certain databases (currently PostgreSQL, MariaDB, and SQLite
  3.35+). On other databases, it will not be set.
* It does not work with many-to-many relationships.

> * If the model has :attr:`~django.db.models.Options.order_with_respect_to` set,

Reviewer: I wonder if, as a bugfix, this should be omitted in the docs and only mentionned in the release notes.
We noticed that the model.save() documentation does not refer to this feature.

The release note will provide enough guidance for anyone interested in removing a workaround.

Last, if you decide to keep this info, I would suggest to move it from the caveats since this section is for unexpected behaviours.

Reviewer 2: I think I agree with @donadef here
We have two options:
- add documentation around how `_order` is automatically populated in general, such as on save, rather than only specific notes for bulk_create and add `.. versionchanged: 6.0` notes
- remove the docs and release on it being populated for bulk_create

I personally don't think the docs are needed

Author: Thanks for the discussion. I’ve removed the changes from both options.txt and querysets.txt, and left only the note in releases/6.0.txt as suggested.

I agree this keeps the docs clean while still making the change visible for those affected.

Let me know if anything else should be adjusted!

>   the ``_order`` field will be  automatically populated for all objects based

Reviewer 3: ```suggestion
  the ``_order`` field will be automatically populated for all objects based
```

Author: thank you, I fixed it

> * If the model has ``order_with_respect_to`` set, the ``_order`` field will be
>   automatically populated for all created objects based on existing objects in
>   the database.

Reviewer 4: ```suggestion
* If the model has :attr:`~django.db.models.Options.order_with_respect_to` set,
the ``_order`` field will be  automatically populated for all objects based
on existing objects in the database in the order they are provided.
```

We'll also need a `.. versionadded 6.0` below to denote this was recently changed and also a mention in the 6.0 release notes.

Author: Thank you for the comment. I have made the changes as suggested. However, are you also asking me to add the 6.0 documentation under the /docs/releases path?

Reviewer 3: Yes, I think that's what Simon is requesting.
